model/atmosphere/dycore/tests/dycore_stencil_tests/test_compute_horizontal_advection_term_for_vertical_velocity.py

```diff
@@ -13,9 +13,21 @@
     compute_horizontal_advection_term_for_vertical_velocity,
 )
 from icon4py.model.common import dimension as dims
+from icon4py.model.common.grid import horizontal as h_grid, icon as icon_grid
+from icon4py.model.common.type_alias import vpfloat, wpfloat
 from icon4py.model.common.utils.data_allocation import random_field, zero_field
 from icon4py.model.testing.helpers import StencilTest
-from icon4py.model.common.type_alias import vpfloat, wpfloat
+
+
+def _horizontal_range(grid):
+    if isinstance(grid, icon_grid.IconGrid):
+        # For the ICON grid we use the proper domain bounds (otherwise we will run into non-protected skip values)
+        edge_domain = h_grid.domain(dims.EdgeDim)
+        return grid.start_index(edge_domain(h_grid.Zone.LATERAL_BOUNDARY_LEVEL_7)), grid.end_index(
+            edge_domain(h_grid.Zone.HALO)
+        )
+    else:
+        return 0, gtx.int32(grid.num_edges)
```

Comment:
This is a new pattern and breaks the grid abstraction. Maybe all grids should provide markers, but for simple grid we set them to 0...num_x everywhere?